app/src/main/java/com/gathering/android/event/repo/ApiEventRepository.kt

```diff
@@ -1,17 +1,65 @@
 package com.gathering.android.event.repo
 
+import android.content.Context
 import com.gathering.android.common.BODY_WAS_NULL
 import com.gathering.android.common.RESPONSE_IS_NOT_SUCCESSFUL
 import com.gathering.android.common.ResponseState
+import com.gathering.android.common.createRequestPartFromUri
+import com.gathering.android.common.requestBody
 import com.gathering.android.event.model.EventModel
+import com.gathering.android.event.putevent.repo.PutEventModel
+import dagger.hilt.android.qualifiers.ApplicationContext
+import okhttp3.RequestBody
 import retrofit2.Call
 import retrofit2.Callback
 import retrofit2.Response
 import javax.inject.Inject
 
 class ApiEventRepository @Inject constructor(
     private val eventRemoteService: EventRemoteService,
+    @ApplicationContext private val context: Context,
 ) : EventRepository {
+    override suspend fun addEvent(event: PutEventModel) {
+        val eventName: RequestBody = event.eventName.requestBody()
+        val eventDescription: RequestBody = event.description.requestBody()
+        val latitude = event.lat.requestBody()
+        val longitude = event.lon.requestBody()
+        val dateTime = event.dateAndTime.requestBody()
+        val attendees = event.getAttendeesJson().requestBody()
+        val filePart =
+            context.createRequestPartFromUri(event.photoUri) ?: throw IllegalStateException()
+
+        eventRemoteService.addEvent(
+            eventName = eventName,
+            eventDescription = eventDescription,
+            latitude = latitude,
+            longitude = longitude,
+            dateTime = dateTime,
+            attendees = attendees,
+            photo = filePart
+        )
+    }
+
+    override suspend fun editEvent(event: PutEventModel) {
+        val eventId: RequestBody = event.eventId.requestBody()
+        val eventName: RequestBody = event.eventName.requestBody()
+        val eventDescription: RequestBody = event.description.requestBody()
+        val latitude = event.lat.requestBody()
+        val longitude = event.lon.requestBody()
+        val dateTime = event.dateAndTime.requestBody()
+        val attendees = event.getAttendeesJson().requestBody()
+        val filePart = context.createRequestPartFromUri(event.photoUri)
+
+        eventRemoteService.editEvent(eventId = eventId,
+            eventName = eventName,
+            eventDescription = eventDescription,
+            latitude = latitude,
+            longitude = longitude,
+            dateTime = dateTime,
+            attendees = attendees,
+            photo = filePart)
```

Comment:
The formatting here has problem,
Push `ctrl + shift + L` If not answered, please put the closing parentheses in the next line